src/messages/worker.rs

```diff
@@ -6,6 +6,12 @@ use crate::messages::common::{SubworkerDefinition, TaskFailInfo, WorkerConfigura
 use crate::Priority;
 use crate::{OutputId, TaskId, TaskTypeId, WorkerId};
 
+#[derive(Serialize, Deserialize, Debug)]
+pub enum ConnectionRegistration {
+    Worker(RegisterWorker),
+    Custom,
```

Comment:
This seems like a generalization, but realistically you can only use it for one thing at a time, because there are no input parameters to `Custom`, so the `CustomConnectionHandler` can't recognize what process is trying to connect.

Does this need to be inside tako? Can't the workers just connect to the server node outside of tako?

Reply:
Its up to handler to decide what kind of connection it is, e.g. it may wait for the first additional message (as is actually implemented in HQ, streamer is registered with the first message).

The all this is for reusing opened server port, otherwise connecting directly to a new port would be easier and cleaner